[AY1920S1-CS2113-T14-3] WordUp

tessa-z opened the pull request:
@tran-quang-thanh @njw95 @chelsea148629 @kevin996231

leowweixiang commented on src/main/java/Dictionary/Word.java:
```diff
@@ -0,0 +1,76 @@
+     * String represents the closest time that user search
+     * for a specific word
+     */
+    private String closetSearch;
```
closestSearch you mean?
To clarify, closest time -> this is a string containing a time format?

leowweixiang commented on src/main/java/Dictionary/WordBank.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Looks up for meaning of a specific word
```
Good to add here that "if word is not in Wordbank, call OxfordCall to find the word"

leowweixiang commented on src/main/java/command/BadCommand.java:
```diff
@@ -0,0 +1,26 @@
+import storage.Storage;
+import ui.Ui;
+
+/**
```
Add such headers for other command subclasses. Also, if the execute() is nontrivial (long code), do add a header explain what it does and how

leowweixiang commented on src/main/java/command/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+
+    @Override
+    public String execute(Ui ui, WordBank wordBank, Storage storage) {
+        //ask ui to print something
```
Comment different from code. Same thing appears for the others: Try not to copy-paste code/comments

leowweixiang commented on src/main/java/command/HistoryCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public String execute(Ui ui, WordBank wordBank, Storage storage) {
+        //ask ui to print something
```
Comment different from code

leowweixiang commented on src/main/java/command/OxfordCall.java:
```diff
@@ -0,0 +1,73 @@
+        return "https://od-api.oxforddictionaries.com/api/v2/entries/" + language + "/" + word_id + "?" + "fields=" + fields + "&strictMatch=" + strictMatch;
+    }
+
+    public static String doInBackground(String word) throws NoWordFoundException {
```
Add header comment for these public methods

leowweixiang commented on src/main/java/command/OxfordCall.java:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+    /**
+     * Extract the first definition inside the metadata
```
Can explain more. How are you extracting the definition - what does the metadata look like and which part are you extracting?

leowweixiang commented on src/main/java/command/QuizCommand.java:
```diff
@@ -0,0 +1,56 @@
+        return false;
+    }
+
+    public void generateQuiz(WordBank wordBank) throws WordBankNotEnoughForQuizException {
```
Header comment. Also, for each rough section of code, explain what you're doing to setup the quiz

leowweixiang commented on src/main/java/command/QuizCommand.java:
```diff
@@ -0,0 +1,56 @@
+
+    public QuizCommand(){
+        this.random = new Random();
+        this.options = new String[4];
```
Good to put this 4 as a private static final attribute. That way you can easily alter it if you want to change to 5-option MCQ etc

leowweixiang commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+                // CREATE A HELP COMMAND TO SHOW THE AVAILABLE INSTRUCTION
+                return null;
+            } else if (taskInfo[0].equals("add")) {
+                if (taskInfo.length == 1) {
```
Will be helpful to have a comment describing the string format you're looking for

leowweixiang commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+                    throw new WrongDeleteFormatException();
+                }
+                String[] wordAndTags = taskInfo[1].split("t/");
+                //delete word
```
Good, short comment that lets others zoom into specific code

leowweixiang commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+                if (taskInfo.length == 1) {
+                    throw new WrongAddFormatException();
+                }
+                String[] wordDetail = taskInfo[1].split("w/");
```
Again consider putting your "w/" etc as private static final attribs

leowweixiang commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+                    return new DeleteCommand(wordDescription, tags);
+                }
+            } else if (taskInfo[0].equals("search")) {
+                if (taskInfo.length == 1 || !taskInfo[1].startsWith("w/")) {
```
Because the each case has like 10+ lines of code, consider putting them in separate private methods

leowweixiang commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,182 @@
+        }
+    }
+
+    public Stack<Word> loadHistoryFromFile() {
```
Header comment. Make explicit the difference between loadHistory and loadWordBank (Stack structure, why)
Also, the code is very similar. Can both be done in one method, or are there times you have to load each separately?

leowweixiang commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,182 @@
+            }
+            return wordBank;
+        } catch (IOException e) {
+            e.printStackTrace();
```
You may want to feedback something more useful (to the user)

leowweixiang reviewed (commented):
General:
In terms of features and code I think y'all are on a good track :) What's lacking are unit tests (important!!) and header comments on public methods.

kevin996231 committed:
optimize search closest command

njw95 committed:
Fix UG adoc style

kevin996231 committed:
Restructure search command

tessa-z committed:
Merge pull request #97 from njw95/master

Updated UserGuide.adoc

kevin996231 committed:
Fix checkstyle

kevin996231 committed:
Add comment and javadoc

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

kevin996231 committed:
Press up button to show previous command in textfield

tessa-z committed:
Create YueHan.adoc

tessa-z committed:
Update YueHan.adoc

tessa-z committed:
Update YueHan.adoc

Commit:
Improve GUI and edit SearchCommand.

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Merge branch 'master' of https://github.com/tessa-z/main

kevin996231 committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

Update instructions for running JAR file from terminal.
Include Schedule and History feature instructions.

kevin996231 committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Delete YueHan.adoc

tessa-z committed:
Merge branch 'master' into master

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

Add search frequency feature instructions.

tran-quang-thanh committed:
Update UserGuide.adoc

Add Instruction for delete tag, and search_begin

Commit:
Improve exception handling for Schedule feature.

Commit:
Merge branch 'master' of https://github.com/tessa-z/main

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Merge branch 'master' into master

tessa-z committed:
Merge pull request #99 from tessa-z/master

Update UserGuide

tessa-z committed:
Merge pull request #98 from kevin996231/master

Update search command

Commit:
Improve exception handling for History feature.

Commit:
Improve trim in Parser class.

njw95 committed:
Update UserGuide.adoc

njw95 committed:
Update UserGuide.adoc

njw95 committed:
Pre-PE

kevin996231 committed:
Update UserGuide.adoc

kevin996231 committed:
Fix issue #120

kevin996231 committed:
Update UserGuide.adoc

kevin996231 committed:
Update UserGuide.adoc

kevin996231 committed:
Update help function

kevin996231 committed:
Fix checkstyle

kevin996231 committed:
Fix quiz bug

kevin996231 committed:
Fix checkstyle

njw95 committed:
Added isAlphabetString checking of word and meaning

njw95 committed:
Fix UG adoc

njw95 committed:
Merge branch 'master' into master

njw95 committed:
Fix Style

njw95 committed:
fix checkstyle

njw95 committed:
Fix style

njw95 committed:
Fix style

njw95 committed:
Fix Style

njw95 committed:
throw exception style issue

chelsea148629 committed:
Merge pull request #124 from kevin996231/master

Fix issues and update help

njw95 committed:
Allow commas , fullstop and ;

njw95 committed:
Check style

tessa-z committed:
Update UserGuide.adoc

chelsea148629 committed:
solve issue #121

chelsea148629 committed:
fix checkstyle

tran-quang-thanh committed:
Fix Command bugs and add spell checking

chelsea148629 committed:
Merge pull request #127 from tessa-z/master

Post mock-pe updates

tran-quang-thanh committed:
Fix PE error

njw95 committed:
Merge branch 'master' into master

tessa-z committed:
Update build.gradle

Commit:
Fix schedule termination bugs.

Commit:
Fix schedule reminder showing previous words bug.

chelsea148629 committed:
Merge pull request #128 from chelsea148629/master

solve issue #121 quiz duplicate issue

chelsea148629 committed:
Merge pull request #126 from njw95/master

Added isAlphabetString to check word and meaning

kevin996231 committed:
Use up/down button to switch between previous/next input

tran-quang-thanh committed:
Fix conflict

tran-quang-thanh committed:
Add excel

kevin996231 committed:
Modify word difference formula

tran-quang-thanh committed:
Add Excel and commands deal with tag

tran-quang-thanh committed:
Add Excel and commands deal with tag and fix style

tran-quang-thanh committed:
Add Excel and commands deal with tag and fix style

tran-quang-thanh committed:
Merge pull request #95 from tran-quang-thanh/master

Add excel, fix PE bug, and add commands with Tag

kevin996231 committed:
Fix CommandTest

kevin996231 committed:
Add search test

kevin996231 committed:
Fix checkstyle

kevin996231 committed:
Merge branch 'master' into master

Commit:
Fix.

Commit:
Resolve merge conflict.

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Fix style.

Commit:
Fix style.

tessa-z committed:
Merge pull request #137 from tessa-z/master

Version that creates text file if it doesn't exist

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

tran-quang-thanh committed:
Remove Storage constructor with excel path, creates excel file when creates data folder

tran-quang-thanh committed:
Merge pull request #138 from tran-quang-thanh/master

Use only 1 constructor for Storage

Commit:
Enable single command input for schedule function. Improve reminder error handling.

Commit:
Fix style.

Commit:
Create remindertext file.

kevin996231 committed:
Modify spell check threshold

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Link list of words to reminder popup.

kevin996231 committed:
Fix bug of failing to edit word in txt file

kevin996231 committed:
Fix checkstyle

Commit:
Load reminders on start from file storage.

Commit:
Fix updateFile filePath mistake when deleting reminder information from file.

kevin996231 committed:
Merge branch 'master' of https://github.com/kevin996231/main

kevin996231 committed:
Modify storage to fit tests

kevin996231 committed:
Modify creating and deleting testfile methods

kevin996231 committed:
Fix checkstyle

kevin996231 committed:
Fix search bug

Commit:
Fix style.

Commit:
Fix test style.

tessa-z committed:
Merge pull request #139 from tessa-z/master

Create single line input following project constraints for schedule feature.

kevin996231 committed:
Merge branch 'master' into master

kevin996231 committed:
Merge pull request #136 from kevin996231/master

Use up/down button to switch between previous/next input

kevin996231 committed:
Fix storage path and search test case

kevin996231 committed:
Merge pull request #141 from kevin996231/master

Fix storage path and search test case

kevin996231 committed:
Revert "Fix storage path and search test case"

njw95 committed:
Draft

kevin996231 committed:
Merge pull request #142 from AY1920S1-CS2113-T14-3/revert-141-master

Revert "Fix storage path and search test case"

kevin996231 committed:
Fix storage path and search test case

kevin996231 committed:
Merge pull request #143 from kevin996231/master

Fix storage path and search test case

tran-quang-thanh committed:
Limit size cell

tran-quang-thanh committed:
Merge https://github.com/AY1920S1-CS2113-T14-3/main

kevin996231 committed:
Fix search bug when bank is empty

kevin996231 committed:
Fix typo

njw95 committed:
Added Union support for chaining synonyms together

njw95 committed:
Fix showSearchSynonym ui

njw95 committed:
Merged with main

Commit:
Add SetReminderCommand test.

tran-quang-thanh committed:
Fix storage

tran-quang-thanh committed:
Merge branch 'master' of https://github.com/njw95/main into test-synonym

Commit:
Fix style.:

tran-quang-thanh committed:
Fix commands for synonym and store synonyms in excel

njw95 committed:
before manual paste

tran-quang-thanh committed:
Merge pull request #147 from tessa-z/master

Add SetReminderCommand test.

njw95 committed:
Copy of synonym excel fixed

Commit:
Minor edit.

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Resolve Storage constructor conflict.

Commit:
Fix list_tags typo.

tran-quang-thanh committed:
Fix UI bugs on search_begin and list_tags

njw95 committed:
Merge pull request #148 from tran-quang-thanh/master

Fix UI bugs on search_begin and list_tags and fix conflict in Storage constructor

njw95 committed:
Merge from main

njw95 committed:
Fix Style

njw95 committed:
Fix style

njw95 committed:
Fix style

Commit:
Fix commandTest fails.

Commit:
Fix SearchTest fails.

njw95 committed:
Merge pull request #150 from njw95/master

Added searchSynonym to help

tessa-z committed:
Merge branch 'master' into master

njw95 committed:
Merge pull request #149 from tessa-z/master

Fix storage constructor

Commit:
Add DeveloperGuide.adoc.

tessa-z committed:
Merge pull request #152 from tessa-z/master

Add DeveloperGuide.adoc.

Commit:
Update DeveloperGuide.adoc with use cases and user stories from GDocs.

tessa-z committed:
Merge pull request #153 from tessa-z/master

Update DeveloperGuide.adoc with use cases and user stories from GDocs.

chelsea148629 committed:
refactor add example

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

chelsea148629 committed:
improve ui

chelsea148629 committed:
fix null comparison in UI

chelsea148629 committed:
DG 4.5 Quiz Figures

chelsea148629 committed:
Update add_example and fix format

chelsea148629 committed:
Update 3.8 quiz

chelsea148629 committed:
fix checkstyle

chelsea148629 committed:
Merge pull request #140 from chelsea148629/master

Add example command

chelsea148629 committed:
QuizSequenceDiagram

chelsea148629 committed:
update images in quiz command

Commit:
Fix style.

tessa-z committed:
Merge branch 'master' into master

tran-quang-thanh committed:
Add search_tag and list_tags to UserGuide

Commit:
Resolve CommandTest merge conflict.

Commit:
Resolve CommandTest merge conflict.

tran-quang-thanh committed:
Update UG DG

tran-quang-thanh committed:
Merge https://github.com/AY1920S1-CS2113-T14-3/main

tran-quang-thanh committed:
Add image for search feature in DG

tran-quang-thanh committed:
update UG DG

tran-quang-thanh committed:
update UG DG

tran-quang-thanh committed:
Merge https://github.com/AY1920S1-CS2113-T14-3/main

njw95 committed:
Update DG adoc

njw95 committed:
Fix DG picture alignment

njw95 committed:
Fix DG adoc

njw95 committed:
Fix DG alignment

njw95 committed:
synonym png issue fix

njw95 committed:
Added search_syn and search_tag to UG

Commit:
Fix style.

Commit:
Merge branch 'master' of https://github.com/tessa-z/main

tessa-z committed:
Update DeveloperGuide.adoc

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Update DevGuide.adoc.

Commit:
Update AboutUs.adoc. Add app screenshot.

tessa-z committed:
Update README.md

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Fix style.

Commit:
Fix style.

Commit:
Fix SetReminderCommandTest exception.

tran-quang-thanh committed:
Merge pull request #155 from njw95/master

Allow commas and fullstop for words and update DG + UG

Commit:
Fix test excel file deletion.

tessa-z committed:
Merge branch 'master' into master

tessa-z committed:
Merge pull request #154 from tessa-z/master

Fix style. Add tests to reminder class.

Commit:
Reupload wordup screenshot.

Commit:
Merge branch 'master' of https://github.com/tessa-z/main

tessa-z committed:
Merge pull request #157 from tessa-z/master

Update documentation

tessa-z committed:
Update README.md

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

kevin996231 committed:
Merge pull request #158 from kevin996231/master

Fix minor bugs and typo

tessa-z committed:
Update schedule instructions

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

kevin996231 committed:
Update DeveloperGuide.adoc

kevin996231 committed:
Update UserGuide.adoc

kevin996231 committed:
Add photos for command example

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

kevin996231 committed:
Merge pull request #159 from kevin996231/master

Add pictures

chelsea148629 committed:
Ui.png

chelsea148629 committed:
No screenshot image on project page. issue #151

kevin996231 committed:
Update UserGuide.adoc

kevin996231 committed:
Fix typo

tessa-z committed:
Add RecentlyAddedSequenceDiagram.png

tessa-z committed:
Update DeveloperGuide.adoc

tessa-z committed:
Update DeveloperGuide.adoc

tessa-z committed:
Add manual testing instructions

tessa-z committed:
Update DeveloperGuide.adoc

Commit:
Rename classes.

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update UserGuide.adoc

tran-quang-thanh committed:
Merge pull request #160 from tran-quang-thanh/master

Add pictures for tags command in UG and search feature in DG

tessa-z committed:
Update UserGuide.adoc

tessa-z committed:
Update DeveloperGuide.adoc

tessa-z committed:
Update DeveloperGuide.adoc

kevin996231 committed:
Add more test

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

tran-quang-thanh committed:
Merge pull request #161 from kevin996231/master

Add more test

njw95 committed:
Added synonym pictures to UG

njw95 committed:
Fix picture

njw95 committed:
Fix PNG

chelsea148629 committed:
Update quiz command

njw95 committed:
Rename synonym pic

chelsea148629 committed:
Update add example

njw95 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

tessa-z committed:
Update DeveloperGuide.adoc

tessa-z committed:
Update DeveloperGuide.adoc

njw95 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

Commit:
Update.

kevin996231 committed:
fix bugs

kevin996231 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-3/main

njw95 committed:
Remove diagrams

njw95 committed:
Merge pull request #163 from njw95/master

Remove non-linked diagrams

Commit:
Final update.

Commit:
Resolve merge conflict.
